pkg/policy/rule.go

```diff
@@ -191,5 +230,55 @@ func (r *rule) canReach(ctx *SearchContext, state *traceState) api.Decision {
 		}
 	}
 
+	return entitiesDecision
+}
+
+func (r *rule) canReachEntities(ctx *SearchContext, state *traceState) api.Decision {
+	entitiesFound := false
+	for _, r := range r.Egress {
+		for _, entity := range r.ToEntities {
+			entitiesFound = true
+			l, ok := api.EntitySelectorMapping[entity]
+			if !ok {
+				ctx.PolicyTrace("     unsupported entity found: %s, skipping", entity)
+				continue
+			}
+
+			if l.Matches(ctx.To) {
+				ctx.PolicyTrace("+     Found all required labels to match entity %s\n", entity)
+				return api.Allowed
+			}
+		}
+	}
+
+	if entitiesFound {
+		return api.Denied
```

Comment:
I don't think this is correct. This will stop processing in the middle of the policy tree if any rule contains an egress section with a ToEntities which does not match the current context. The default deny decision has to happen at the very end with the following logic:

allowed if allowed is returns
allowed if undecided and policy enforcement on { ingress | egress } is disabled
denied if undecided and policy enforcement on { ingress | egress } is enabled

Reply:
Yes, I was writing this with an assumption that if the egress rule to entity is set, we should deny all traffic that doesn't go to this entity. Will remove.